### PR TITLE
Disallow adding a destructive, or realtime, effect which has been deleted since application start up.

### PR DESCRIPTION
**Pull request by ksoze95**
Resolves: #3719 #3817 

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [ ] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/RealtimeEffectPanel.cpp [hunk 1]
```diff
@@ -1270,6 +1271,15 @@ class RealtimeEffectListWindow : public wxScrolledWindow
       if(effectID.empty())
          return;
 
+      auto plug = PluginManager::Get().GetPlugin(effectID);
+      if(!plug)
+         return;
+
+      if(!PluginManager::IsPluginAvailable(*plug)) {
+         wxMessageBox("This plugin could not be loaded.\nIt may have been deleted.", "Plugin Error");
```
Comment (vsverchinsky) [inline src/RealtimeEffectPanel.cpp:1279]: Please make these strings translatable

File: src/effects/EffectUI.cpp [hunk 2]
```diff
@@ -1231,6 +1232,12 @@ DialogFactoryResults EffectUI::DialogFactory(wxWindow &parent,
    if (!plug)
       return false;
 
+   if (!PluginManager::IsPluginAvailable(*plug))
+   {
+      wxMessageBox("This plugin could not be loaded.\nIt may have been deleted.", "Plugin Error");
```
Comment (vsverchinsky) [inline src/effects/EffectUI.cpp:1237]: Same

**Review by vsverchinsky — CHANGES_REQUESTED** [review]
Dialog message text needs translation

File: src/RealtimeEffectPanel.cpp (hunk 1 above)
Comment (ksoze95) [reply src/RealtimeEffectPanel.cpp]: Done.

File: src/effects/EffectUI.cpp (hunk 2 above)
Comment (ksoze95) [reply src/effects/EffectUI.cpp]: Done.

**Review by vsverchinsky — APPROVED** [review]

File: src/RealtimeEffectPanel.cpp [hunk 3]
```diff
@@ -1270,6 +1270,15 @@ class RealtimeEffectListWindow : public wxScrolledWindow
       if(effectID.empty())
          return;
 
+      auto plug = PluginManager::Get().GetPlugin(effectID);
+      if(!plug)
+         return;
+
+      if(!PluginManager::IsPluginAvailable(*plug)) {
+         wxMessageBox(_("This plugin could not be loaded.\nIt may have been deleted."), _("Plugin Error"));
```
Comment (Paul-Licameli) [inline src/RealtimeEffectPanel.cpp:1278]: Use ::AudacityMessageBox

File: src/effects/EffectUI.cpp [hunk 4]
```diff
@@ -1231,6 +1231,12 @@ DialogFactoryResults EffectUI::DialogFactory(wxWindow &parent,
    if (!plug)
       return false;
 
+   if (!PluginManager::IsPluginAvailable(*plug))
+   {
+      wxMessageBox(_("This plugin could not be loaded.\nIt may have been deleted."), _("Plugin Error"));
```
Comment (Paul-Licameli) [inline src/effects/EffectUI.cpp:1236]: Use ::AudacityMessageBox

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]

File: src/effects/EffectUI.cpp (hunk 4 above)
Comment (Paul-Licameli) [reply src/effects/EffectUI.cpp:1236]: Or better, use BasicUI::ShowMessageBox in both places.

File: src/RealtimeEffectPanel.cpp (hunk 3 above)
Comment (ksoze95) [reply src/RealtimeEffectPanel.cpp]: Replaced wxMessageBox with BasicUI::ShowMessageBox as suggested below.

File: src/menus/FileMenus.cpp [hunk 5]
```diff
@@ -184,6 +186,41 @@ void OnOpen(const CommandContext &context )
...
+            }
+         });
+   }
+
+   if (unavailablePlugins > 0)
+   {
+      BasicUI::ShowMessageBox(
+         XO("This project contains some plugins that cannot be found on the system."),
+         BasicUI::MessageBoxOptions()
+            .Caption(XO("Project Error")));
```
Comment (LWinterberg) [inline src/menus/FileMenus.cpp:222]: Let's do
> Window title: Missing Plugins
> This project contains some realtime effect plugins that cannot be found on this system.
> The project may sound different than intended. [Learn more](https://audacityteam.org/errors#missing-plugins)
> [OK]

with the "learn more" having an accessible link to https://audacityteam.org/errors#missing-plugins

**Review by LWinterberg — APPROVED** [review]

File: src/effects/EffectUI.cpp [hunk 6]
```diff
@@ -1243,6 +1244,16 @@ DialogFactoryResults EffectUI::DialogFactory(wxWindow &parent,
    if (!plug)
       return false;
 
+   if (!PluginManager::IsPluginAvailable(*plug))
```
Comment (Paul-Licameli) [inline src/effects/EffectUI.cpp:1247]: Should this message be shown also in case `plug` is null?

**Review by Paul-Licameli — APPROVED** [review]
A few suggestions to improve it, but nothing that should block it.

**Pull request merged**